scala/x509.scala

```diff
@@ -0,0 +1,44 @@
+import org.mongodb.scala._
+import scala.collection.JavaConverters._
+import com.mongodb.connection.ClusterSettings
+import com.mongodb.connection.SslSettings
+import org.mongodb.scala.bson.collection.mutable.Document
+import org.mongodb.scala.bson.BsonString
+import com.mongodb.client.model.FindOneAndUpdateOptions
+
+object MongoDBx509 extends App {
+   System.setProperty("javax.net.ssl.trustStore","client.truststore");
+   System.setProperty("javax.net.ssl.trustStorePassword","password");
+
+   System.setProperty("javax.net.ssl.keyStore", "client.keystore");
+   System.setProperty("javax.net.ssl.keyStorePassword", "password");
+
+   // null value will force mongoDB server to use DN from cert
+   val credential = MongoCredential.createMongoX509Credential(null);
+
+   val settings = MongoClientSettings.builder()
+      .credential(credential)
+      .applyToClusterSettings(
+         (builder: ClusterSettings.Builder)
+         => builder.hosts(List(new ServerAddress("localmongo1", 27017)).asJava))
+      .applyToSslSettings(
+         (builder : SslSettings.Builder)
+         => builder.enabled(true))
+      .build();
+
+   val client : MongoClient = MongoClient(settings);
+
+   val db : MongoDatabase = client.getDatabase("test")
+   val collection : MongoCollection[org.mongodb.scala.bson.collection.mutable.Document] = db.getCollection("stuff")
```

Comment:
Is there a namespace collision that requires you to have the full class name for `Document`? If not, you should be able to use `Document` since you imported it on line 5.

Reply:
Both mutable and immutable Document classes have the class name "Document", just in different packages. While it's not necessary to spell this out completely here (since I only imported the mutable Document) depending on your import setup you might end up with an immutable document instead if you just use "Document". Particularly since there are snippets and might not include imports, I felt it was better to use a more verbose classname here to prevent ambiguity particularly since Scala's error messages if you DO use the immutable Document type by mistake do not lend themselves to easy debugging of the problem.

TL;DR I'd like it to stay exactly as is. This was very intentional.